src/abstractprobprog.jl

```diff
@@ -80,3 +81,29 @@ end
 function Base.rand(model::AbstractProbabilisticProgram)
     return rand(Random.default_rng(), NamedTuple, model)
 end
+
+"""
+    predict(
+        [rng::AbstractRNG=Random.default_rng(),]
+        [T=NamedTuple,]
+        model::AbstractProbabilisticProgram,
+        params,
+    ) -> T
+
+Draw a sample from the joint distribution specified by `model` conditioned on the values in
+`params`.
+
+The sample will be returned as format specified by `T`.
+"""
+function StatsBase.predict(rand::AbstractRNG, T::Type, model::AbstractProbabilisticProgram, params)
+    return rand(rng, T, condition(model, params))
```

Comment:
The docstring for `condition` would technically limit `params` to observations, for which this definitely does the wrong thing. Should `condition` be generalized?

Reply:
> The docstring for condition would technically limit params to observations

What do you mean by this?

Reply:
The docstring for `condition` reads:
> Condition the generative model `model` on some observed data

So as documented it is only meant for conditioning on observations, or equivalently, leaves in a DAG. But here `predict` wants to fix everything _but_ the leaves in the DAG (or perhaps more generally, a subgraph that contains a leaf).